src/main/kotlin/com/coder/gateway/views/CoderGatewayRecentWorkspaceConnectionsView.kt

```diff
@@ -206,94 +199,42 @@ class CoderGatewayRecentWorkspaceConnectionsView(private val setContentCallback:
-                                                        try {
-                                                            client.stopWorkspace(workspace)
-                                                            fetchWorkspaces()
-                                                        } catch (e: Exception) {
-                                                            logger.error("Could not stop workspace ${workspace.name}", e)
-                                                        }
-                                                    }
-                                                }
-                                        }
-                                    }
-                                },
-                            ).applyToComponent { isEnabled = workspaceWithAgent?.workspace?.latestBuild?.status == WorkspaceStatus.RUNNING }
-                                .gap(RightGap.SMALL)
-                            actionButton(
-                                object : DumbAwareAction(
-                                    CoderGatewayBundle.message("gateway.connector.recent-connections.terminal.button.tooltip"),
-                                    "",
-                                    CoderIcons.OPEN_TERMINAL,
-                                ) {
-                                    override fun actionPerformed(e: AnActionEvent) {
-                                        withoutNull(workspaceWithAgent, deployment?.client) { ws, client ->
-                                            val link = client.url.withPath("/me/${ws.name}/terminal")
-                                            BrowserUtil.browse(link.toString())
-                                        }
-                                    }
-                                },
-                            )
                         }.topGap(gap)
-                        if (deploymentError == null || showError) {
+
+                        connections.forEach { workspaceProjectIDE ->
+                            val enableLinks = listOf(WorkspaceStatus.STOPPED, WorkspaceStatus.CANCELED, WorkspaceStatus.FAILED, WorkspaceStatus.STARTING, WorkspaceStatus.RUNNING).contains(workspaceWithAgent?.workspace?.latestBuild?.status)
+                            val inLoadingState = listOf(WorkspaceStatus.STARTING, WorkspaceStatus.CANCELING, WorkspaceStatus.DELETING, WorkspaceStatus.STOPPING).contains(workspaceWithAgent?.workspace?.latestBuild?.status)
+
                             row {
-                                // There must be a way to make this properly wrap?
-                                label("<html><body style='width:350px;'>" + status.third + "</html>").applyToComponent {
-                                    foreground = status.second
+                                if (inLoadingState) {
+                                    icon(AnimatedIcon.Default())
+                                }
```

Comment:
Not at all a blocker and completely optional, more just idly musing, but I wonder if we want to turn the pair back into a triple and put a nullable icon in there, which would let us also show a spinner while the query is running as well (like it used to) and bring back the error icon for API errors (if we want, honestly the red color is probably sufficient).

Reply:
I think adding the error icon back is a good idea and fits in this PR :)